helix-view/src/document.rs

```diff
@@ -955,6 +958,35 @@ impl Document {
         }
     }
 
+    #[cfg(unix)]
+    // Detect if the file is readonly and change the readonly field if necessary (unix only)
+    pub fn detect_readonly(&mut self) -> bool {
```

Comment:
These functions don't need to return `self.readonly`. The return value isn't used anywhere and the field is already public on the doc so callers can check there.

Reply:
I believe this function also needs to be called in `Document::open` with the other auto-detection functions. That way the read-only indicator will be accurate if you `:open some-read-only-file.txt`, choose one from the picker, etc..

Reply:
I have `detect_readonly()` running every time `set_path()` is called, and `set_path()` is called inside `Document::open`, so I don't think we need to.